lms/djangoapps/discussion_api/tests/test_api.py

```diff
@@ -1273,6 +1274,56 @@ def test_basic(self, mock_emit):
             }
         )
 
+    @ddt.data(
+        *itertools.product(
+            [
+                FORUM_ROLE_ADMINISTRATOR,
+                FORUM_ROLE_MODERATOR,
+                FORUM_ROLE_COMMUNITY_TA,
+                FORUM_ROLE_STUDENT,
+            ],
+            [True, False],
+            [True, False],
+            ["no_group", "group_none", "group_set"],
+        )
+    )
+    @ddt.unpack
+    def test_group_id(self, role_name, course_is_cohorted, topic_is_cohorted, data_group_state):
+        cohort_course = CourseFactory.create(
+            discussion_topics={"Test Topic": {"id": "test_topic"}},
+            cohort_config={
+                "cohorted": course_is_cohorted,
+                "cohorted_discussions": ["test_topic"] if topic_is_cohorted else [],
+            }
+        )
+        CourseEnrollmentFactory.create(user=self.user, course_id=cohort_course.id)
+        cohort = CohortFactory.create(course_id=cohort_course.id, users=[self.user])
```

Comment:
minor: would it make more sense to put this line behind `if course_is_cohorted` ?